src/api/app/controllers/webui/comments_controller.rb

```diff
@@ -67,13 +73,13 @@ def permitted_params
     params.require(:comment).permit(:body, :parent_id)
   end
 
-  def find_commentable
-    commentable = [Project, Package, BsRequest].find { |klass| klass.name == params[:commentable_type] }
+  def set_commentable_type
+    @commentable_type = [Project, Package, BsRequest].find { |klass| klass.name == params[:commentable_type] }
 
-    if commentable.nil?
-      redirect_to(root_path, error: 'Failed to create comment')
-    else
-      @commented = commentable.find(params[:commentable_id])
-    end
+    raise ArgumentError, "Invalid commentable #{params[:commentable_type]} supplied." unless @commentable_type
+  end
+
+  def find_commentable
+    @commented = @commentable_type.find_by(id: params[:commentable_id]) unless @commentable_type.nil?
```

Comment:
This method is called `find_commentable`, but it sets the instance variable `@commented`. It's a bit confusing. We could also merge `set_commentable_type` and `find_commentable`. Having two callbacks with such a minimal amount of code isn't needed. One would be enough.

Reply:
Right, done.